src/backend/common/helpers/matchstats_helper.py

```diff
@@ -72,184 +191,51 @@ def build_Minv_matrix(
                     team1_id = team_id_map[team1[3:]]
                     for team2 in alliance_teams:
                         M[team1_id, team_id_map[team2[3:]]] += 1
+
         return np.linalg.pinv(M)
 
     @classmethod
-    def build_s_matrix(
+    def __build_s_matrix(
         cls,
         matches: List[Match],
         team_id_map: TTeamIdMap,
-        stat_type: StatType,
-        init_stats: Optional[Dict[StatType, Dict[TeamId, int]]] = None,
-        init_stats_default: int = 0,
-        limit_matches: Optional[int] = None,
-    ) -> "np.ndarray[int]":
+        point_accessor: Callable[[Match, AllianceColor], float],
+    ) -> "np.ndarray[float]":
         n = len(team_id_map.keys())
         s = np.zeros((n, 1))
         for match in matches:
             if match.comp_level != CompLevel.QM:  # only consider quals matches
                 continue
 
-            treat_as_unplayed = (
-                limit_matches is not None and match.match_number > limit_matches
-            )
-
```

Comment:
I'm assuming this was removed because we never passed `limit_matches`, so all of the downstream code can be removed?

Reply:
Correct, yeah. I can put it back in if you'd want it for the future, but it's never used anywhere.